tsdb/head_append.go

```diff
@@ -265,7 +267,25 @@ func (a *headAppender) Append(ref storage.SeriesRef, lset labels.Labels, t int64
 		}
 	}
 
+	// (Q from cstyan) this block will result in whatever the last seen metadata
+	// for a series (based on labelset hash) was being used as the metadata for
+	// an entire block, is that right?
+	hasNewMetadata := false
+
 	s.Lock()
+	if s.typ != meta.Type {
+		hasNewMetadata = true
+		s.typ = meta.Type
+	}
+	if s.unit != meta.Unit {
+		hasNewMetadata = true
+		s.unit = meta.Unit
+	}
+	if s.help != meta.Help {
+		hasNewMetadata = true
+		s.help = meta.Help
+	}
```

Comment:
Maybe the in-memory series should store the model/metadata type instead of keeping each field separately?